include/fastdds/rtps/transport/ChainingTransport.h

```diff
@@ -0,0 +1,260 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CHAINING_TRANSPORT_H
+#define CHAINING_TRANSPORT_H
+
+#include "TransportInterface.h"
+#include "ChainingTransportDescriptor.h"
+
+#include <map>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+
+class ChainingReceiverResource;
+
+/**
+ * This is the base class for chaining adapter transports.
+ *    - Directly proxies all operations except Send and Receive
+ *
+ *    - Has a pointer to the low level transport
+ * @ingroup TRANSPORT_MODULE
+ */
+class ChainingTransport : public TransportInterface
+{
+
+public:
+
+    RTPS_DllAPI
+    ChainingTransport(
+            const ChainingTransportDescriptor& t)
+        : TransportInterface(0)
+        , low_level_transport_(t.low_level_descriptor->create_transport())
+    {
+        transport_kind_ = low_level_transport_->kind();
+    }
+
+    RTPS_DllAPI
+    virtual ~ChainingTransport()
+    {
+    }
+
+    RTPS_DllAPI
+    bool init(
+            const fastrtps::rtps::PropertyPolicy* properties = nullptr) override
+    {
+        return low_level_transport_->init(properties);
+    }
+
+    //! Checks whether there are open and bound sockets for the given port.
+    RTPS_DllAPI
+    bool IsInputChannelOpen(
+            const fastrtps::rtps::Locator_t& loc) const override
+    {
+        return low_level_transport_->IsInputChannelOpen(loc);
+    }
+
+    //! Checks for low level support.
+    RTPS_DllAPI
+    bool IsLocatorSupported(
+            const fastrtps::rtps::Locator_t& loc) const override
+    {
+        return low_level_transport_->IsLocatorSupported(loc);
+    }
+
+    /**
+     * Converts a given remote locator (that is, a locator referring to a remote
+     * destination) to the main local locator whose channel can write to that
+     * destination.
+     */
+    RTPS_DllAPI
+    fastrtps::rtps::Locator_t RemoteToMainLocal(
+            const fastrtps::rtps::Locator_t& loc) const override
+    {
+        return low_level_transport_->RemoteToMainLocal(loc);
+    }
+
+    /**
+     * Starts listening on the specified locator.
+     */
+    RTPS_DllAPI
+    bool OpenInputChannel(
+            const fastrtps::rtps::Locator_t& loc,
+            TransportReceiverInterface* receiver_interface,
+            uint32_t max_message_size) override;
+
+    /**
+     * Opens a socket on the given address and port (as long as they are white listed).
+     */
+    RTPS_DllAPI
+    bool OpenOutputChannel(
+            SendResourceList& sender_resource_list,
+            const fastrtps::rtps::Locator_t& loc) override;
+
+    //! Removes the listening socket for the specified port.
+    RTPS_DllAPI
+    bool CloseInputChannel(
+            const fastrtps::rtps::Locator_t& loc) override
+    {
+        return low_level_transport_->CloseInputChannel(loc);
+    }
+
+    RTPS_DllAPI
```

Comment:
I am missing the doxygen documentation of several of the public methods. It is true that the API reference currently does not include the transport layer, but I suggest having the methods already documented. At least the new pure virtual methods `send` and `receive` should be documented. The other methods are overrides and the user could look into the base methods for more information (if they are documented).

Reply:
Done